modules/govuk_containers/manifests/frontend/haproxy.pp

```diff
@@ -0,0 +1,64 @@
+# == Class: Govuk_containers::Frontend::Haproxy
+#
+# Manage Haproxy frontend for container boxes
+#
+# === Parameters:
+#
+# [*app_domain*]
+#   Global environment domain 
+#
+# [*wildcard_publishing_certificate*]
+#   Wildcard publishing certificate
+#
+# [*wildcard_publishing_key*]
+#   Wildcard publishing key
+#
+class govuk_containers::frontend::haproxy (
+  $app_domain,
+  $wildcard_publishing_certificate,
+  $wildcard_publishing_key,
+) {
+
+  include ::haproxy
+
+  apt::pin { 'haproxy':
+    packages => 'haproxy',
+    release  => 'trusty-backports',
+    priority => 1001, # 1001 will cause a downgrade if necessary
+  }
+
+  file { [ '/etc/haproxy/ssl' ]:
+    ensure => 'directory',
+  }
+
+  file { '/etc/haproxy/ssl/publishing.pem':
+    ensure  => 'present',
+    owner   => haproxy,
+    mode    => '0640',
+    content => inline_template("${wildcard_publishing_certificate}${wildcard_publishing_key}"),
+    require => File['/etc/haproxy/ssl'],
+  }
+
+  haproxy::frontend { $::hostname:
+    mode    => 'http',
+    bind    => {
+      "${::ipaddress_eth0}:443" => ['ssl', 'crt', '/etc/haproxy/ssl/publishing.pem'],
+    },
+    options => {
+      'timeout client' => '30s',
+      'option'         => [
+        'httplog',
+        'accept-invalid-http-request',
+      ],
+      'use_backend'    => '%[req.hdr(host),lower,map(/etc/haproxy/domains-to-backends.map,bk_default)]',
+    },
+  }
+
+  haproxy::mapfile { 'domains-to-backends':
+    ensure   => 'present',
+    mappings => [
```

Comment:
Could this be defined in hiera in the future? Just thinking about rapid adoption of all our apps as containers, and the less per app config we can add the better (IMO).